viz-lib/src/visualizations/table/Renderer.jsx

```diff
@@ -51,7 +52,33 @@ function SearchInputInfoIcon({ searchColumns }) {
   );
 }
 
-const SearchInput = React.forwardRef(({ searchColumns, ...props }, ref) => {
+// Some explanation why this weird solution with `clearCallbackRef` is used.
+// When visualization options are editing, renderer may re-create table columns quite often.
+// Search input is one of table columns, and it does not depend on options.
+// We could just render it once and then leave it for React, but there is a feature
+// that needs access to Search input's value: when SOME options or data columns change - search
+// input should reset. If we'll use managed input's state in renderer itself - it will create
```

Comment:
If we don't clear it, will it filter the updated table?

Reply:
Do you mean when query returns new data? Well, it _should_. Currently it has behavior fully kept from Angular version: when data changes - search should be reset. In this case it's an optimization to update/reset it only when relevant part of data changes, otherwise it will re-render entire table on every typed character.

Reply:
We should just keep the filtering on even if the data (or other options) change. This will simplify the code here and probably provide a more expected experience for the user.

Reply:
Okay. Should I implement it in this PR or better open a new one?

Reply:
It basically replaces this one, so whatever makes more sense to you.

Reply:
@arikfr  https://github.com/getredash/redash/pull/4899/commits/f2358e6904251e0f3f29e5bd494168177fb5f6e2